grakn-engine/src/main/java/ai/grakn/engine/printer/JacksonPrinter.java

```diff
@@ -0,0 +1,117 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.engine.printer;
+
+import ai.grakn.concept.Concept;
+import ai.grakn.engine.controller.response.Answer;
+import ai.grakn.engine.controller.response.ConceptBuilder;
+import ai.grakn.graql.Printer;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * <p>
+ *     This class is used to convert the responses from graql queries into objects which can be Jacksonised into their
+ *     correct Json representation.
+ * </p>
+ *
+ * @author Filipe Peliz Pinto Teixeira
+ */
+public class JacksonPrinter implements Printer<Object>{
+    private Optional<Integer> limitEmbedded = Optional.empty();
+    private static ObjectMapper mapper = new ObjectMapper();
+
+    private JacksonPrinter(){
+
+    }
+
+    private JacksonPrinter(int limitEmbedded){
+        this.limitEmbedded = Optional.of(limitEmbedded);
+    }
+
+    public static JacksonPrinter create(){
+        return new JacksonPrinter();
+    }
+
+    public static JacksonPrinter create(int limitEmbedded){
+        return new JacksonPrinter(limitEmbedded);
+    }
+
+    @Override
+    public String build(Object object) {
+        try {
+            return mapper.writeValueAsString(object);
+        } catch (IOException e) {
+            throw new RuntimeException(String.format("Error during serialising {%s}", object), e);
+        }
+    }
+
+    @Override
+    public Object graqlString(boolean inner, Concept concept) {
+        return ConceptBuilder.build(concept);
+    }
+
+    @Override
+    public Object graqlString(boolean inner, ai.grakn.graql.admin.Answer answer) {
+        return Answer.create(answer);
+    }
+
+    @Override
+    public Object graqlString(boolean inner, boolean bool) {
+        return bool;
+    }
+
+    @Override
+    public Object graqlStringDefault(boolean inner, Object object) {
+        return object;
+    }
+
+    @Override
+    public Object graqlString(boolean inner, Map map) {
+        Stream<Map.Entry> entries = map.<Map.Entry>entrySet().stream();
+        if(limitEmbedded.isPresent()) entries = entries.limit(limitEmbedded.get());
+
+        return entries.collect(Collectors.toMap(
+                entry -> graqlString(inner, entry.getKey()),
+                entry -> graqlString(inner, entry.getKey())
+        ));
+    }
+
+    @Override
+    public Object graqlString(boolean inner, Collection collection) {
+        Stream stream = collection.stream();
+        if(limitEmbedded.isPresent()) stream = stream.limit(limitEmbedded.get());
```

Comment:
`limitEmbedded` here will mean we limit the number of results in queries and also do other bad stuff (like truncating the returned path in `compute path`). I'm pretty sure that's not what we want: @marco-scoppetta is probably the person to ask about what exactly `limitEmbedded` should do.

Reply:
@aelred is right, don't use it to limit results, they should be already limited because the limit x; is already specified in the graql string. `limitEmbedded` was used for limiting the "embedded" nodes we were embedding in a response e.g. if user asks for a relationship instance which has thousands of roleplayers (IHS Markit) we would limit the number of roleplayers to return in a single HAL object)

Reply:
Alright I will kill it for the moment. Thanks guys!